### DISCUSSION TITLE
nodeOrigin on a per Node basis

### DISCUSSION THREAD

**Discussion by cmd8** [xyflow/xyflow#2871] [Q&A]
Is it possible to specify the origin for each node individually or at least for each node type, as the title suggests?

For example, I have a node that expands upward with animation based on a certain condition. However, there are other types of nodes that must expand downward. Is it possible to achieve this?

**Comment by moklick** [comment]
It's not possible with the current version but I get your use case and we will check if we can implement this feature.

**Reply by cmd8** [reply]
> It's not possible with the current version but I get your use case and we will check if we can implement this feature.

Thank you for the reply. If I were to look into implementing this where should I start?

**Reply by moklick** [reply]
You can only do this in the library itself. You would need to check every place where `nodeOrigin` is being used and then use `node.nodeOrigin || nodeOrigin`.

**Reply by cmd8** [reply]
Thanks! Please review the PR when you have time. Tested in the Drag and Drop example, works perfectly fine - https://github.com/wbkd/react-flow/pull/2874
